movementlabsxyz/movement
# Indexer: Add Indexer and Hasura GraphQl to movement

# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.


Add the indexer and Hasura GraphQl engine to process compose.

This PR depends on these one:
 * https://github.com/movementlabsxyz/aptos-core/pull/53
 * https://github.com/movementlabsxyz/aptos-indexer-processors/pull/2
# Changelog


Create an Indexer crate under network/suzuka.
Update process compose to run Indexer and Hasura.
Remove the indexer started with Suzuka node.
Create 2 process-compose script: indexer and Hasura
The Postgres Db is now started by the indexer and not Suzuka-node.

# Testing
If there's some issue for building (dieser error) you need to clean the repo with `cargo clean`
If you have the issue with fixed crate that need rustc version 1.79.0 execute this command: `cargo update -p fixed --precise 1.27.0`
To run everything:
```
CELESTIA_LOG_LEVEL=FATAL nix develop --extra-experimental-features nix-command --extra-experimental-features flakes --command bash  -c "just suzuka-full-node native build.setup.eth-local.celestia-local.indexer.hasura --keep-tui"
```
When the Hasura server is started, open a browser at this url: `http://localhost:8085/console`
To add the Suzuka Tx db to Hasura:
 * go in Data (top of the windows)
 * Connect database
 * Connect existing database
 * Put a name to `Database name`
 * Select database Url
 * set Database URL to `postgresql://postgres:password@database:5432/postgres`
 * Validate with connect dabase.
 * Under Datamanager select your database-> public
 * All the indexer table appears. Select track on the transactions table for example.
 * Go under API
 * Type this query
 ```
query MyQuery {
  transactions {
    hash
    success
    type
    version
  }
}
```
To see all Tx in the Db.


# Outstanding issues
I start only one indexer processor: dedfault_processor. I'm going to see how to start more processor to populate more data in the db.
Available processors are:
```
    AccountTransactionsProcessor,
    AnsProcessor(AnsProcessorConfig),
    CoinProcessor,
    DefaultProcessor,
    EventsProcessor,
    FungibleAssetProcessor,
    MonitoringProcessor,
    NftMetadataProcessor(NftMetadataProcessorConfig),
    ObjectsProcessor(ObjectsProcessorConfig),
    StakeProcessor(StakeProcessorConfig),
    TokenProcessor(TokenProcessorConfig),
    TokenV2Processor(TokenV2ProcessorConfig),
    TransactionMetadataProcessor,
    UserTransactionProcessor,
    ParquetDefaultProcessor(ParquetDefaultProcessorConfig),
    ParquetFungibleAssetProcessor(ParquetFungibleAssetProcessorConfig),
```
